beetsplug/edit.py

```diff
@@ -216,20 +239,26 @@ def edit(self, album, objs, fields):
           everything).
         """
         # Present the YAML to the user and let her change it.
-        success = self.edit_objects(objs, fields)
+        if album:
+            success = self.edit_objects(objs, None, fields)
+        else:
+            success = self.edit_objects(objs, fields, None)
 
         # Save the new data.
         if success:
             self.save_changes(objs)
 
-    def edit_objects(self, objs, fields):
+    def edit_objects(self, objs, item_fields, album_fields):
```

Comment:
While I'm not *thrilled* about the need to introduce two field sets here, I can see that it might be necessary. (Previously, it was nice that this method just worked on Model objects abstractly; it didn't know or care about Item and Album specifically.)